### PR TITLE
handle RPC execution error

### LINKED ISSUE #3603: Unhandled exception error with no details
I've noticed some logs like this:
2022-03-17 23:32:34.706+00:00 | vert.x-eventloop-thread-2 | ERROR | RoutingContext | Unhandled exception in router

There's no error details and can be confusing to node operators (is it critical or not? how do we solve it?)

Unclear how to replicate at this stage we might wanna look in where the exception is logged instead.

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
* call handleJsonRpcError in one extra spot
* and some extra logging about where the error occurred

## Fixed Issue(s)
fixes #3603 

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java [hunk 1]
```diff
@@ -58,11 +58,13 @@ public static Handler<RoutingContext> handler(
                   } catch (IOException e) {
                     final String method = executor.getRpcMethodName(ctx);
                     LOG.error("{} - Error streaming JSON-RPC response", method, e);
-                    throw new RuntimeException(e);
+                    handleJsonRpcError(ctx, null, RpcErrorType.INTERNAL_ERROR);
```
Comment (garyschulte) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java:61]: This seems like a good practice, but afaict IOException is going to be thrown by execute in the same thread and the runtime exception would be handled by the outer try/catch.  So I don't think it is strictly necessary.

The change seems ok, but I am curious if Devin gave any justification

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java [hunk 2]
```diff
@@ -58,11 +58,13 @@ public static Handler<RoutingContext> handler(
...
                   }
                 },
                 () -> handleJsonRpcError(ctx, null, RpcErrorType.PARSE_ERROR));
       } catch (final RuntimeException e) {
+        final String method = ctx.get(ContextKey.REQUEST_BODY_AS_JSON_OBJECT.name());
+        LOG.error("Unhandled exception in JSON-RPC executor for method {}", method, e);
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java:67]: this change will log the method that was called so we're getting extra context

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java (hunk 1 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java:61]: think you're right on this one, we're not strictly gaining anything except uniformity in the code

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/handlers/JsonRpcExecutorHandler.java:61]: from Devin - It replaces throwing a new RuntimeException with a call to a method named handleJsonRpcError, which includes the error message from the exception.

**Review by usmansaleem — APPROVED** [review]
LGTM

**Pull request merged**